node/common/transport/amqp/lib/amqp_receiver.js

```diff
@@ -55,7 +56,8 @@ AmqpReceiver.prototype._onAmqpMessage = function (message) {
      * @event module:azure-iot-amqp-base.AmqpReceiver#message
      * @type {AmqpMessage}
      */
-  this.emit('message', message);
+  var msg = new Message(message.body);
+  this.emit('message', msg);
```

Comment:
We want the message that is emitted to be transport-agnostic. we can either duck-type AmqpMessage to meet the Message interface or convert it to a Message class. TBD if it's the role of the transport or the client directly to do it.